### PR TITLE
Make neo skeleta be auxiliary keys

### PR DESCRIPTION
**Pull request by sffc**
Replaces #4778 and #4779

See discussion on #4779

### CODE REVIEW

**Comment by sffc** [comment]
Some initial size results, the first time when these numbers include the lookup table:

Common configurations:

- --keys 'datetime/patterns/gregory/date_day_skeleton@1' 'datetime/patterns/gregory/date_ext_skeleton@1'
- --format blob2

| --locales | --deduplication-strategy | Environment | Size |
|---|---|---|---|
| full | none | Combined (Packed) | 151824 B |
| full | none | Split (Aux Keys) | 115985 B |
| recommended | none | Combined (Packed) | 124817 B |
| recommended | none | Split (Aux Keys) | 91117 B |
| recommended | maximal | Combined (Packed) | 121951 B |
| recommended | maximal | Split (Aux Keys) | 69724 B |

**Comment by sffc** [comment]
Question: I could put all of the skeleta in the same key, or I could split them into 3 keys. The advantage of more keys is that static slicing can work slightly better:

- Date includes Date and Day skeleta
- Time includes Time skeleta
- DateTime includes Day and Time skeleta

The disadvantage is that there are more lookup tables, which wastes space, especially in non-deduplication aka hybrid mode (and probably base languages mode). You can see the impact of that in my previous post.

**Comment by sffc** [comment]
I'm currently leaning toward a single key because even though it increases the default size, it reduces the floor, which is what matters for our most highly-constrained clients.

**Comment by sffc** [comment]
Thought: We could even stuff the length patterns into this same key and just make one big but efficient "get your pattern here" key.

As a reminder, display names remain separate.

**Comment by sffc** [comment]
No use reviewing commit-by-commit because there is a lot of intermediate state that I rolled back. It's not that big of a change if you ignore the datagen'd files.

File: components/datetime/src/provider/neo.rs [hunk 1]
```diff
@@ -588,25 +604,24 @@ impl AsULE for SkeletonDataIndex {
...
-            index,
         }
     }
 }
 
 #[icu_provider::data_struct(
     marker(
         GregorianDateNeoSkeletonPatternsV1Marker,
-        "datetime/patterns/gregory/date_skeleton@1"
+        "datetime/patterns/gregory/skeleton@1"
```
Comment (Manishearth) [inline components/datetime/src/provider/neo.rs:624]: question (non blocking): I forget; is there a reason this is gregorian-only? Do we plan to add more or is this now shared between calendars?

**Review by Manishearth — APPROVED** [review]

File: components/datetime/src/provider/neo.rs (hunk 1 above)
Comment (sffc) [reply components/datetime/src/provider/neo.rs:624]: I plan to add `"datetime/patterns/buddhist/skeleton@1"`, ... using all of our existing calendar-specific-key infrastructure

**Pull request merged**